frontend/tests/README.md

```diff
@@ -0,0 +1,361 @@
+  - Redirect to GitHub authorization
+  - Successful callback handling
+  - localStorage population
+- **Password Authentication** - Tests email/password login
+  - Successful login
+  - Failed login with error messages
+  - Form validation
+- **Session Persistence** - Tests session management
+  - localStorage persistence
+  - Session across page reloads
+- **Logout** - Tests logout functionality
+  - localStorage clearing
+  - API failure handling
+- **UI State Changes** - Tests UI updates based on login state
+  - Login button visibility when logged out
+  - User menu visibility when logged in
+
+### `logo.spec.ts` - Basic UI Test
+
+Simple test that verifies the Nyrkiö logo renders on the front page.
+
+## Running Tests
+
+### Prerequisites
+
+```bash
+# Install dependencies (if not already installed)
+npm install
+
+# Install Playwright browsers
+npx playwright install chromium
+```
+
+### Run Unit Tests (Mocked API)
+
+```bash
+# Run all unit tests
+npm run test
+
+# Run specific unit test file
+npm run test:unit
+
+# Run with UI mode
+npm run test-ui
+```
+
+### Run Integration Tests (Real API)
+
+**Prerequisites:** MongoDB running, backend configured
+
+```bash
+# Run all integration tests
+npm run test:integration
+
+# Run with UI mode
+npm run test-ui:integration
+
+# View integration test report
+npm run test-report:integration
+```
+
+**See [Integration Tests README](./integration/README.md) for detailed setup.**
+
+### Run Specific Test File
+
+```bash
+npx playwright test tests/auth.spec.ts
+```
+
+### Run Tests in UI Mode (Interactive)
+
+```bash
+npm run test-ui
+```
+
+This opens Playwright's interactive UI where you can:
+- See tests running in real-time
+- Debug failing tests
+- Inspect DOM snapshots
+- View network requests
+
+### Run Single Test
+
+```bash
+npx playwright test tests/auth.spec.ts -g "should display login page"
+```
+
+### Run Tests in Specific Browser
+
+```bash
+# Chromium (default)
+npx playwright test --project=chromium
+
+# Firefox
+npx playwright install firefox
+npx playwright test --project=firefox
+
+# WebKit (Safari)
+npx playwright install webkit
+npx playwright test --project=webkit
+```
+
+### Debug Tests
+
+```bash
+# Run tests in debug mode
+npx playwright test tests/auth.spec.ts --debug
+
+# Run with headed browser (visible)
+npx playwright test tests/auth.spec.ts --headed
+```
+
+## Test Reports
+
+After running tests, view the HTML report:
+
+```bash
+npm run test-report
+```
+
+This opens an interactive HTML report showing:
+- Test results
+- Screenshots on failure
+- Execution timeline
+- Network activity
+
+## Configuration
+
+Test configuration is in `playwright.config.ts`:
+
+- **Base URL:** `http://127.0.0.1:5173`
+- **Web Server:** Automatically starts `npm run dev` before tests
+- **Reporters:**
+  - List (console output)
+  - JSON (machine-readable results)
+  - HTML (interactive report)
+  - Custom Nyrkiö reporter (sends results to Nyrkiö API)
```

Comment:
This should be migrated to nyrkio/change-detection action but can be separate PR.

Actually, you can disable it here if you want. It might be outdated and at least doesn't handle PRs well. Might create a mess.